src/SSW.Rewards.Application/Quizzes/Commands/AddNewQuiz/AddNewQuizCommandValidator.cs

```diff
@@ -0,0 +1,22 @@
+﻿namespace SSW.Rewards.Application.Quizzes.Commands.AddNewQuiz;
+public class AddNewQuizCommandValidator : AbstractValidator<AddNewQuizCommand>
+{
+    private readonly IApplicationDbContext _context;
+
+    public AddNewQuizCommandValidator(IApplicationDbContext context)
+    {
+        _context = context;
+
+        RuleFor(c => c)
+            .MustAsync(BeUniqueQuiz);
+    }
+
+    private async Task<bool> BeUniqueQuiz(AddNewQuizCommand command, CancellationToken cancellationToken)
+    {
+        return await _context.Quizzes
+            .AnyAsync(q => 
+                q.Id == command.Quiz.QuizId
+                ||
```

Comment:
Remove these lines. We will only search on the quiz name.
Include a check for Archived, so we allow a quiz to be created with the same name as long as any existing records are archived.

Reply:
Done